internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -0,0 +1,572 @@
+			ForceNew:     true,
+		},
+
+		"source_location": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			StateFunc:        location.StateFunc,
+			DiffSuppressFunc: location.DiffSuppressFunc,
+			ValidateFunc:     validation.StringIsNotEmpty,
+			RequiredWith:     []string{"source_server_id"},
+		},
+
+		"source_server_id": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"administrator_password": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Sensitive:    true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			RequiredWith: []string{"administrator_username", "administrator_password"},
+		},
+
+		"compute_tier": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"Free",
+				"M25",
+				"M30",
+				"M40",
+				"M50",
+				"M60",
+				"M80",
+			}, false),
+		},
+
+		"high_availability_mode": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				// Confirmed with service team the `SameZone` is currently not supported.
+				string(mongoclusters.HighAvailabilityModeDisabled),
+				string(mongoclusters.HighAvailabilityModeZoneRedundantPreferred),
+			}, false),
+		},
+
+		"public_network_access_enabled": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"storage_size_in_gb": {
+			Type:         schema.TypeInt,
+			Optional:     true,
+			ValidateFunc: validation.IntBetween(32, 16384),
+		},
+
+		"tags": commonschema.Tags(),
+
+		"version": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"5.0",
+				"6.0",
+				"7.0",
+			}, false),
+		},
+	}
+}
+
+func (r MongoClusterResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{}
+}
+
+func (r MongoClusterResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MongoCluster.MongoClustersClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var state MongoClusterResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := mongoclusters.NewMongoClusterID(subscriptionId, state.ResourceGroupName, state.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameter := mongoclusters.MongoCluster{
+				Location:   azure.NormalizeLocation(state.Location),
```

Comment:
This function is deprecated, please use `location.Normalize`
```suggestion
				Location:   location.Normalize(state.Location),
```